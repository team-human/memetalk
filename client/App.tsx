import * as React from 'react'
import * as eva from '@eva-design/eva';
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs'
import TopicScreen from './Pages/topic'
import ProfileScreen from './Pages/profile'
import { apolloClient } from './Apollo/ApolloClient'
import { ApolloProvider } from '@apollo/client'
import { ApplicationProvider} from '@ui-kitten/components';
import ComposeScreen from './Pages/compose';
import LoginScreen from './Pages/login';


const TopicStack = createStackNavigator()

function TopicStackScreen() {
  return (
    <TopicStack.Navigator>
      <TopicStack.Screen name="Topic" component={TopicScreen} />
    </TopicStack.Navigator>
  )
}

const ProfileStack = createStackNavigator()

function ProfileStackScreen() {
  return (
    <ProfileStack.Navigator>
      <ProfileStack.Screen name="Profile" component={ProfileScreen} />
    </ProfileStack.Navigator>
  )
}

const ComposeStack = createStackNavigator()

function ComposeStackScreen() {
  return (
    <ComposeStack.Navigator>
      <ComposeStack.Screen name="Compose" component={ComposeScreen}/>
    </ComposeStack.Navigator>
  )
}

const Stack = createStackNavigator();
const Tab = createBottomTabNavigator()

export default function App() {
  return (
    <ApplicationProvider {...eva} theme={eva.light}>
    <ApolloProvider client={apolloClient}>
      <NavigationContainer>
      <Stack.Navigator initialRouteName="Home">
        <Stack.Screen name="Home" component={LoginScreen} />
      </Stack.Navigator>
        {/* <Tab.Navigator>
          <Tab.Screen name="Login" component={LoginScreen} />
          <Tab.Screen name="Topic" component={TopicStackScreen} />
          <Tab.Screen name="Profile" component={ProfileStackScreen} />
          <Tab.Screen name="Compose" component={ComposeStackScreen}/>
        </Tab.Navigator> */}
      </NavigationContainer>
    </ApolloProvider>
    </ApplicationProvider>

  )
}
