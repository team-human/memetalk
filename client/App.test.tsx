import React from 'react';
import { render } from 'react-native-testing-library';

import App from './App';

describe('<App />', () => {
  it("App", async () => {
    const component = render(<App />);
    expect(component.toJSON()).toMatchSnapshot();
  }); 
});

