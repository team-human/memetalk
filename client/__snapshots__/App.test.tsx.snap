// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            null,
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Array [
                    Object {
                      "zIndex": 1,
                    },
                    false,
                  ]
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  importantForAccessibility="auto"
                  onLayout={[Function]}
                  pointerEvents="box-none"
                  style={null}
                >
                  <View
                    pointerEvents="box-none"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "zIndex": 0,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "rgb(255, 255, 255)",
                          "borderBottomColor": "rgb(216, 216, 216)",
                          "flex": 1,
                          "shadowColor": "rgb(216, 216, 216)",
                          "shadowOffset": Object {
                            "height": 0.5,
                            "width": 0,
                          },
                          "shadowOpacity": 0.85,
                          "shadowRadius": 0,
                        }
                      }
                    />
                  </View>
                  <View
                    pointerEvents="box-none"
                    style={
                      Object {
                        "height": 64,
                        "maxHeight": undefined,
                        "minHeight": undefined,
                        "opacity": undefined,
                        "transform": undefined,
                      }
                    }
                  >
                    <View
                      pointerEvents="none"
                      style={
                        Object {
                          "height": 20,
                        }
                      }
                    />
                    <View
                      pointerEvents="box-none"
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                          "flexDirection": "row",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <View
                        pointerEvents="box-none"
                        style={
                          Object {
                            "marginHorizontal": 16,
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          accessibilityRole="header"
                          numberOfLines={1}
                          onLayout={[Function]}
                          style={
                            Object {
                              "color": "rgb(28, 28, 30)",
                              "fontSize": 17,
                              "fontWeight": "600",
                            }
                          }
                        >
                          Home
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  />
                  <View
                    accessibilityElementsHidden={false}
                    closing={false}
                    gestureVelocityImpact={0.3}
                    importantForAccessibility="auto"
                    onClose={[Function]}
                    onGestureBegin={[Function]}
                    onGestureCanceled={[Function]}
                    onGestureEnd={[Function]}
                    onOpen={[Function]}
                    onTransitionStart={[Function]}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                    transitionSpec={
                      Object {
                        "close": Object {
                          "animation": "spring",
                          "config": Object {
                            "damping": 500,
                            "mass": 3,
                            "overshootClamping": true,
                            "restDisplacementThreshold": 10,
                            "restSpeedThreshold": 10,
                            "stiffness": 1000,
                          },
                        },
                        "open": Object {
                          "animation": "spring",
                          "config": Object {
                            "damping": 500,
                            "mass": 3,
                            "overshootClamping": true,
                            "restDisplacementThreshold": 10,
                            "restSpeedThreshold": 10,
                            "stiffness": 1000,
                          },
                        },
                      }
                    }
                  >
                    <View
                      pointerEvents="box-none"
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        forwardedRef={[Function]}
                        onGestureHandlerEvent={[Function]}
                        onGestureHandlerStateChange={[Function]}
                        style={
                          Object {
                            "flex": 1,
                            "transform": Array [
                              Object {
                                "translateX": 0,
                              },
                              Object {
                                "translateX": 0,
                              },
                            ],
                          }
                        }
                      >
                        <View
                          pointerEvents="none"
                          style={
                            Object {
                              "backgroundColor": "rgb(242, 242, 242)",
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "shadowColor": "#000",
                              "shadowOffset": Object {
                                "height": 1,
                                "width": -1,
                              },
                              "shadowOpacity": 0.3,
                              "shadowRadius": 5,
                              "top": 0,
                              "width": 3,
                            }
                          }
                        />
                        <View
                          style={
                            Array [
                              Object {
                                "flex": 1,
                                "overflow": "hidden",
                              },
                              Array [
                                Object {
                                  "backgroundColor": "rgb(242, 242, 242)",
                                },
                                undefined,
                              ],
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "flexDirection": "column-reverse",
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "flex": 1,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "flex": 1,
                                    "justifyContent": "center",
                                  }
                                }
                              >
                                <Text>
                                  Home screen
                                </Text>
                                <View
                                  accessibilityRole="button"
                                  accessibilityState={Object {}}
                                  accessible={true}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "opacity": 1,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {},
                                      ]
                                    }
                                  >
                                    <Text
                                      style={
                                        Array [
                                          Object {
                                            "color": "#007AFF",
                                            "fontSize": 18,
                                            "margin": 8,
                                            "textAlign": "center",
                                          },
                                        ]
                                      }
                                    >
                                      Go to Details
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderTopColor": "rgb(216, 216, 216)",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "elevation": 8,
          "height": 49,
          "left": 0,
          "paddingBottom": 0,
          "paddingHorizontal": 0,
          "position": null,
          "right": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityLabel="Home, tab, 1 of 2"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                Object {
                  "color": "rgb(0, 122, 255)",
                },
                Object {
                  "fontSize": 10,
                },
                undefined,
              ]
            }
          >
            Home
          </Text>
        </View>
        <View
          accessibilityLabel="Settings, tab, 2 of 2"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                Object {
                  "color": "#8E8E8F",
                },
                Object {
                  "fontSize": 10,
                },
                undefined,
              ]
            }
          >
            Settings
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
